fn/option.go

```diff
@@ -213,3 +216,12 @@ func (o Option[A]) UnsafeFromSome() A {
 	}
 	panic("Option was None()")
 }
+
+// String returns a string representation of the Option.
+func (o Option[A]) String() string {
+	if o.isSome {
+		return fmt.Sprintf("%v", o.some)
```

Comment:
We should indicate an option value here and print "Some(...)"

Reply:
The loggings are mostly for users, and I think this will confuse them.

Reply:
I am not a fan of altering the string render functions for these general types because users may not understand them. They don't need to understand them. Clarity is more important here, imo.

Reply:
I don't think we need to indicate it, otherwise, we might as well not have this `String` since it's already printed in logs as `bool, value`. Adding a `Some` doesn't help IMO, as we already know it's `none` when it's empty, the rest we care about is its value. That's also my understanding of `Option`, that it's a way of specifying "value not being set".
And nope, I think it's very important that users (devs) understand the logs.

Reply:
You only know it's none when it's empty. But it's only ever full or empty so you have no way of knowing that it could be empty. I'm fine with having a string function. I'm not fine with it omitting the `Some` constructor if the value is present.